core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -173,6 +172,6 @@ private String contextualNameOf(Item i) {
     }
 
     private static boolean startsWithImpl(String str, String prefix, boolean ignoreCase) {
-        return ignoreCase ? StringUtils.startsWithIgnoreCase(str, prefix) : str.startsWith(prefix);
+        return ignoreCase ? str.toLowerCase().startsWith(prefix.toLowerCase()) : str.startsWith(prefix);
```

Comment:
I couldn't easily prove to myself that `prefix` is always non-null here. Am I missing something? If not, then I think we should handle a null `prefix` to minimize risk of regression.

Reply:
we already use `str.startsWith(token)` when `ignoreCase` equals `false`.
`startsWith` method args value must not null.